server: first pass at a debug range page

BramGruneir opened the pull request:
This only includes very basic details at the moment. The next step is to add raft details.

<img width="1369" alt="screen shot 2017-02-14 at 18 11 52" src="https://cloud.githubusercontent.com/assets/1614265/22953920/71221054-f2e1-11e6-9d17-f27ddbb3d02f.png">

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/13590)
<!-- Reviewable:end -->

BramGruneir committed:
server: first pass at a debug range page

This only includes very basic details at the moment. The next step is to add raft details.

bdarnell commented:
How does this relate to the existing raft/range debug pages (especially with #13587 which lets you narrow down to a single range)? It seems like it would be better to improve what's there than to start from scratch, but I'm not sure where you're intending to take this in the future. 

---

Reviewed 11 of 11 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[pkg/server/status.go, line 146 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd21_pHElZzG0H1fShm:-Kd21_pHElZzG0H1fShn:b-8iv6qq) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L146)):*
> ```Go
> 
> func (s *statusServer) parseRangeID(rangeIDParam string) (roachpb.RangeID, error) {
> 	if len(rangeIDParam) == 0 || localRE.MatchString(rangeIDParam) {
> ```

What does a range ID of "local" mean? I don't think it makes sense here; we can leave that to fail with "range id could not be parsed".

---

*[pkg/server/status.go, line 740 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd228m0hZLAKiQHw-3F:-Kd228m0hZLAKiQHw-3G:b-sp1sxh) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L740)):*
> ```Go
> 	existingReplicaIDs := make(map[roachpb.ReplicaID]struct{})
> 	// TODO(bram): Parallelize this to speed it up for large clusters.
> 	for nodeID, alive := range s.nodeLiveness.GetLivenessMap() {
> ```

This is going to be pretty slow on large clusters, and not always necessary. We should probably default to looking up the range descriptor in the meta ranges and querying ranges listed in the descriptor, and only iterate across *all* ranges when specifically requested. 

---

*[pkg/server/status.go, line 741 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd22XIuxpM-Eb-MMqGB:-Kd22XIuxpM-Eb-MMqGC:b-nxxi3p) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L741)):*
> ```Go
> 	// TODO(bram): Parallelize this to speed it up for large clusters.
> 	for nodeID, alive := range s.nodeLiveness.GetLivenessMap() {
> 		if !alive {
> ```

This should probably still count as a Failure.

---

*[pkg/server/status.go, line 944 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd239idLO08opvVDnep:-Kd239idLO08opvVDneq:byjz1kq) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L944)):*
> ```Go
>         {{end}}
>       </DIV>
>       {{if not (eq (len $.NotFound) 0)}}
> ```

It seems strange to list all the nodes where a range is not found (since that is most nodes in a large cluster). I would just list the nodes that have replicas or could not be reached.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-:-Kd24EVYTfewof4TYTuK:b-s8qa2a)*
<!-- Sent from Reviewable.io -->

petermattis commented:
I'm expecting this to replace the existing `_status/{ranges,raft}` pages. Those pages are raw dumps of information. Any per-range information that is on those pages should be present on this new page and hopefully displayed in a friendlier format. 

---

Review status: all files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

*[pkg/server/status.go, line 589 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd2Vhyu7EcckYrgzNR6:-Kd2Vhyu7EcckYrgzNR7:b-1dbsk7) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L589)):*
> ```Go
> 
> 	output := serverpb.RangeResponse{
> 		Details: make([]serverpb.RangeDetail, 0, s.stores.GetStoreCount()),
> ```

You can leave `Details` uninitialized. I don't think this optimization is necessary here.

---

*[pkg/server/status.go, line 603 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd2W-u9aNdmjYLL6z0g:-Kd2W-u9aNdmjYLL6z0h:bmlwni) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L603)):*
> ```Go
> 			detail.Desc = *replica.Desc()
> 		}
> 		output.Details = append(output.Details, detail)
> ```

Sort of strange that you return a result for every store. I would have thought you would only return a result for a store that either had the replica or reported an error.

---

*[pkg/server/status.go, line 740 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd228m0hZLAKiQHw-3F:-Kd2UMq5bTSfsq8AEW4A:b1yeyc7) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L740)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

This is going to be pretty slow on large clusters, and not always necessary. We should probably default to looking up the range descriptor in the meta ranges and querying ranges listed in the descriptor, and only iterate across *all* ranges when specifically requested. 
</blockquote></details>

Sending a request to 10, 100 or even 1000 servers doesn't seem terribly problematic for an operation that is used for debugging. I'd rather get more info when debugging. The big slowness here is the lack of parallelism, but that is easily fixed. See `adminServer.TableStats`.

---

*[pkg/server/status.go, line 944 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd239idLO08opvVDnep:-Kd2V2q-IQYdVG181B7i:bsivml2) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L944)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

It seems strange to list all the nodes where a range is not found (since that is most nodes in a large cluster). I would just list the nodes that have replicas or could not be reached.
</blockquote></details>

Ditto (I already mentioned this in person).

---

*[pkg/server/serverpb/status.proto, line 101 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd2avP5PoBhnP1bmyqk:-Kd2avP5PoBhnP1bmyql:bq5et2g) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/serverpb/status.proto#L101)):*
> ```protobuf
>    ];
>    string err = 3;
>    roachpb.RangeDescriptor desc = 4 [(gogoproto.nullable) = false];
> ```

Should this be `RangeInfo`? Or should the other 3 fields of `RangeDetail` be folded into `RangeInfo`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-:-Kd2_va29x2El7dkWGZm:b-jtexm6)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
I'll be rolling in the results of the calls from the narrowed down raft/range endpoints into this.  And there will be a significant amount of aggregation within handleDebugRange.

---

Review status: 9 of 11 files reviewed at latest revision, 7 unresolved discussions.

---

*[pkg/server/status.go, line 146 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd21_pHElZzG0H1fShm:-Kd6yo3zQjF3W2V8FHE4:bfcqqov) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L146)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

What does a range ID of "local" mean? I don't think it makes sense here; we can leave that to fail with "range id could not be parsed".
</blockquote></details>

Agreed, removed.

---

*[pkg/server/status.go, line 589 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd2Vhyu7EcckYrgzNR6:-Kd6z0YLVW93MBij_15D:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L589)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

You can leave `Details` uninitialized. I don't think this optimization is necessary here.
</blockquote></details>

Done

---

*[pkg/server/status.go, line 603 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd2W-u9aNdmjYLL6z0g:-Kd75fh_T0gzVQsMi8VO:b2z8g60) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L603)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Sort of strange that you return a result for every store. I would have thought you would only return a result for a store that either had the replica or reported an error.
</blockquote></details>

I wanted an exhaustive list of stores.  But that was when I was including a list of stores without any knowledge about the range.  Removed.

---

*[pkg/server/status.go, line 740 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd228m0hZLAKiQHw-3F:-Kd79qSn0zKvz61GbIes:b-a9ug8n) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L740)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Sending a request to 10, 100 or even 1000 servers doesn't seem terribly problematic for an operation that is used for debugging. I'd rather get more info when debugging. The big slowness here is the lack of parallelism, but that is easily fixed. See `adminServer.TableStats`.
</blockquote></details>

I bit the bullet and aded the parallelization and removed the todo.

---

*[pkg/server/status.go, line 741 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd22XIuxpM-Eb-MMqGB:-Kd79zmuvW9iOBEYra3D:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L741)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

This should probably still count as a Failure.
</blockquote></details>

Done.

---

*[pkg/server/status.go, line 944 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd239idLO08opvVDnep:-Kd7MALd62tpkVJLQ4J3:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/status.go#L944)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Ditto (I already mentioned this in person).
</blockquote></details>

Done.

---

*[pkg/server/serverpb/status.proto, line 101 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd2avP5PoBhnP1bmyqk:-Kd7NGahsF4gDDXgo2mh:ba87p7) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/serverpb/status.proto#L101)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Should this be `RangeInfo`? Or should the other 3 fields of `RangeDetail` be folded into `RangeInfo`?
</blockquote></details>

Plan here is to keep adding what's needed to RangeDetail.  RaftState is next.  I didn't want to change the existing RangeInfo results.  

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-:-Kd6yU7Mtt2iPbhNKRtf:b-nniabq)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Updated view:
<img width="1405" alt="screen shot 2017-02-16 at 14 47 03" src="https://cloud.githubusercontent.com/assets/1614265/23038118/d9e52d8a-f456-11e6-9ad9-032e8a4931f1.png">

petermattis commented:
Review status: 9 of 11 files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

*[pkg/server/serverpb/status.proto, line 101 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd2avP5PoBhnP1bmyqk:-KdBtTIiIOd2n-UGWsVV:b5dq79t) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/serverpb/status.proto#L101)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

Plan here is to keep adding what's needed to RangeDetail.  RaftState is next.  I didn't want to change the existing RangeInfo results.  
</blockquote></details>

Ok, though `RangeInfo` already contains `RaftState` and other stuff, so you're probably just re-doing work. 

---

*[pkg/server/serverpb/status.proto, line 111 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd8vMRuCSn1AIt92ToS:-Kd8vMRuCSn1AIt92ToT:b-re0apf) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/serverpb/status.proto#L111)):*
> ```protobuf
> 
> message RangeRequest {
>   // TODO(tamird): use [(gogoproto.customname) = "RangeID"] below. Need to
> ```

@tamird Are we ever going to do this, or are we going to continue copying and pasting this comment everywhere?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 9 of 11 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-:-KdBtbKN09rGwZdFkPrz:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 8 of 11 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 16 unresolved discussions, all commit checks successful.

---

*[pkg/server/server.go, line 353 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdByWPm2Xd0y9AugEz4:-KdByWPnG0nKyUTKqKOK:b-yeaoew) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/server.go#L353)):*
> ```Go
> 	s.admin = newAdminServer(s)
> 	s.status = newStatusServer(s.cfg.AmbientCtx, s.db, s.gossip, s.recorder, s.nodeLiveness,
> 		s.rpcContext, s.node.stores, s.stopper)
> ```

if it's all the same to you, could you write this with each argument on its own line?
```
s.status = newStatusServer(
	s.cfg.AmbientCtx,
	s.db,
	s.gossip,
	s.recorder,
	s.nodeLiveness,
	s.rpcContext,
	s.node.stores,
	s.stopper,
)
```

---

*[pkg/server/status.go, line 151 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBylwIr6yf3vXr2dqO:-KdBylwIr6yf3vXr2dqP:b-ott0w3) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L151)):*
> ```Go
> func (s *statusServer) parseRangeID(rangeIDParam string) (roachpb.RangeID, error) {
> 	if len(rangeIDParam) == 0 {
> 		return 0, fmt.Errorf("no range id provided")
> ```

errors.New?

---

*[pkg/server/status.go, line 155 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBynWVztCpeODCplAQ:-KdBynWVztCpeODCplAR:b-wj5cf6) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L155)):*
> ```Go
> 	id, err := strconv.ParseInt(rangeIDParam, 10, 64)
> 	if err != nil {
> 		return 0, fmt.Errorf("range id could not be parsed: %s", err)
> ```

errors.Wrap or errors.Wrapf

---

*[pkg/server/status.go, line 595 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdByz-wxhMNLddY-JdP:-KdByz-wxhMNLddY-JdQ:bvkcxsz) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L595)):*
> ```Go
> 		replica, err := store.GetReplica(rangeID)
> 		if err != nil {
> 			return nil
> ```

why doesn't this return err?

---

*[pkg/server/status.go, line 723 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBz4eiP51s37v7kf2z:-KdBz4eiP51s37v7kf3-:b-rnknw1) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L723)):*
> ```Go
> // specific range.
> func (s *statusServer) handleDebugRange(w http.ResponseWriter, r *http.Request) {
> 	ctx := s.AnnotateCtx(context.TODO())
> ```

r.Context() instead of context.TODO.

---

*[pkg/server/status.go, line 728 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBz7igszPqYN1hxygC:-KdBz7igszPqYN1hxygD:b8xtui7) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L728)):*
> ```Go
> 	rangeID, err := parseInt64WithDefault(rangeIDString, 1)
> 	if err != nil {
> 		http.Error(w, err.Error(), http.StatusInternalServerError)
> ```

you need to return here.

---

*[pkg/server/status.go, line 744 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBzBKvI5J4h8ttBwly:-KdBzBKvI5J4h8ttBwlz:b-zap6pm) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L744)):*
> ```Go
> 	aliveNodes := len(s.nodeLiveness.GetLivenessMap())
> 	responses := make(chan nodeResponse)
> 	nodeCtx, cancel := context.WithTimeout(context.Background(), base.NetworkTimeout)
> ```

why doesn't this use `ctx`?

---

*[pkg/server/status.go, line 750 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBzJ2SvcI_iEI3RFdb:-KdBzJ2SvcI_iEI3RFdc:bu685xf) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L750)):*
> ```Go
> 			data.Failures = append(data.Failures, serverpb.RangeDetail{
> 				NodeID: nodeID,
> 				Err:    "node not alive",
> ```

this is a very anemic message. please consider adding some words to explain how this was determined, at least.

---

*[pkg/server/status.go, line 815 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBzotsVOrAjMRZRsLB:-KdBzotsVOrAjMRZRsLC:b-jicpeu) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L815)):*
> ```Go
> 	t, err := template.New("webpage").Parse(debugRangeTemplate)
> 	if err != nil {
> 		fmt.Fprintf(w, err.Error())
> ```

why don't these use http.Error?

---

*[pkg/server/status.go, line 818 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdC-1Fe1ZNgTpVjN8g2:-KdC-1Fe1ZNgTpVjN8g3:b8hi95z) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L818)):*
> ```Go
> 		return
> 	}
> 	if err = t.Execute(w, data); err != nil {
> ```

`err :=`

---

*[pkg/server/serverpb/status.proto, line 100 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdByJ6GA36gdKOiNZ1t:-KdByJ6H__nZYy5BfuEG:b3yxyyw) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/serverpb/status.proto#L100)):*
> ```protobuf
>      (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/roachpb.StoreID"
>    ];
>    string err = 3;
> ```

`string error_message = 3;` - we don't use `err` in protos

---

*[pkg/server/serverpb/status.proto, line 111 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd8vMRuCSn1AIt92ToS:-KdBxqi9C5qnrC9s_h4U:b-6js21a) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/serverpb/status.proto#L111)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

@tamird Are we ever going to do this, or are we going to continue copying and pasting this comment everywhere?
</blockquote></details>

Looks like it's just in this file, so I'm ok with the copy and paste for now. OTOH, https://github.com/cockroachdb/cockroach/pull/13587/files adds a custom name to a proto in this file, so maybe this is already moot?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->

tamird commented:
How about some simple tests?

---

Reviewed 1 of 11 files at r1.
Review status: all files reviewed at latest revision, 17 unresolved discussions, all commit checks successful.

---

*[pkg/server/debug.go, line 71 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdC-EW0OZ6XC5ammuow:-KdC-EW0OZ6XC5ammuox:b-3pbdq0) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d12c5c363a69394a962e73fe9fb9f3b7a34869f/pkg/server/debug.go#L71)):*
> ```Go
> 		// download a gzipped file (Content-type comes back application/x-gzip).
> 		w.Header().Add("Content-type", "text/html")
> ```

random change

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-:-KdC-6oovcvxnYLH-gbf:byis3l4)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
*[pkg/server/server.go, line 353 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdByWPm2Xd0y9AugEz4:-KdD0PSfXObBj91yg5WP:b-vga45d) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/server.go#L353)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

if it's all the same to you, could you write this with each argument on its own line?
```
s.status = newStatusServer(
	s.cfg.AmbientCtx,
	s.db,
	s.gossip,
	s.recorder,
	s.nodeLiveness,
	s.rpcContext,
	s.node.stores,
	s.stopper,
)
```
</blockquote></details>

I prefer the single argument per line method as well, but there are others who prefer the bunched up version.  Had to change this back in another PR.  I'm easy going, so either way works for me.  Perhaps we should establish a style guide for this to get rid of any arguments around it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->

tamird commented:
Sad to say: https://github.com/cockroachdb/cockroach/pull/9687

On Fri, Feb 17, 2017 at 4:43 PM, Bram Gruneir <notifications@github.com>
wrote:

> *pkg/server/server.go, line 353 at r2
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdByWPm2Xd0y9AugEz4:-KdD0PSfXObBj91yg5WP:b-vga45d>
> (raw file
> <https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/server.go#L353>):*
> *Previously, tamird (Tamir Duberstein) wrote…*
>
> if it's all the same to you, could you write this with each argument on
> its own line?
>
> s.status = newStatusServer(
> 	s.cfg.AmbientCtx,
> 	s.db,
> 	s.gossip,
> 	s.recorder,
> 	s.nodeLiveness,
> 	s.rpcContext,
> 	s.node.stores,
> 	s.stopper,
> )
>
> I prefer the single argument per line method as well, but there are others
> who prefer the bunched up version. Had to change this back in another PR.
> I'm easy going, so either way works for me. Perhaps we should establish a
> style guide for this to get rid of any arguments around it.
> ------------------------------
>
> *Comments from Reviewable
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/13590>*
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/13590#issuecomment-280774611>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABdsPPJwNoWYTL6dgNA6MRRJFH-QUGhGks5rdhR5gaJpZM4MBHez>
> .
>

petermattis commented:
Review status: all files reviewed at latest revision, 17 unresolved discussions, some commit checks failed.

---

*[pkg/server/server.go, line 353 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdByWPm2Xd0y9AugEz4:-KdD7Afwtht2PPHOhKCs:b-6lwlwh) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/server.go#L353)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

Sad to say: https://github.com/cockroachdb/cockroach/pull/9687
</blockquote></details>

I think it is good practice to not gratuitously change the style of code you're editing. For new code, choose your preference. I will push back hard about establishing a style guide around this. The existing `crlfmt` check is already bordering on too much.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->

tamird commented:
On Fri, Feb 17, 2017 at 5:13 PM, Peter Mattis <notifications@github.com>
wrote:

> I think it is good practice to not gratuitously change the style of code
> you're editing. For new code, choose your preference. I will push back hard
> about establishing a style guide around this. The existing crlfmt check
> is already bordering on too much.
>

The particular instance bram and I were discussing is already changing the
local style.

bdarnell commented:
> I'm expecting this to replace the existing _status/{ranges,raft} pages. Those pages are raw dumps of information. Any per-range information that is on those pages should be present on this new page and hopefully displayed in a friendlier format.

Yeah, the UI side of that page definitely needs to be reworked. But does the server side? The old server side code already had details like parallelism implemented. Can we add more fields to RangeInfo instead of introducing a new RangeDetail? (These names, BTW, are not distinct enough)

---

Review status: all files reviewed at latest revision, 17 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-:-KdDYVhOl9gp3tZ_jvZL:bnpmalu)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
I've added some basic tests and addressed all comments except for the merging of range info and detail.  I'll update this PR by doing that next.

---

Review status: all files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[pkg/server/debug.go, line 71 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdC-EW0OZ6XC5ammuow:-KdD03Kh597lVjaS8_td:bp167nx) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d12c5c363a69394a962e73fe9fb9f3b7a34869f/pkg/server/debug.go#L71)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

random change
</blockquote></details>

added that line back in, must have removed it accidentally.

---

*[pkg/server/server.go, line 353 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdByWPm2Xd0y9AugEz4:-KdIJIuhYrYYoyUaVEAc:bkd1rwb) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/server.go#L353)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

I think it is good practice to not gratuitously change the style of code you're editing. For new code, choose your preference. I will push back hard about establishing a style guide around this. The existing `crlfmt` check is already bordering on too much.
</blockquote></details>

I've made the change.  But as I said, I'm easy going on this one, I'll change it back if there's push back on this.  My preference is to have a more stringent style guide to avoid these types of arguments.  (and the more automated it gets the better)

---

*[pkg/server/status.go, line 151 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBylwIr6yf3vXr2dqO:-KdD3Jib20401iRXK6Qu:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L151)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

errors.New?
</blockquote></details>

Done.

---

*[pkg/server/status.go, line 155 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBynWVztCpeODCplAQ:-KdD3K5eqdu5CIoT_xtf:b-jqdmto) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L155)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

errors.Wrap or errors.Wrapf
</blockquote></details>

done, and also fixed the one in the parsenodeid func above it.

---

*[pkg/server/status.go, line 595 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdByz-wxhMNLddY-JdP:-KdD3zPKBpgk4RheEYEU:b85ecgk) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L595)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

why doesn't this return err?
</blockquote></details>

I've changed the code and added a comment.  Relying on the fact that the error can only be a NewRangeNotFoundError wasn't a good style. 

---

*[pkg/server/status.go, line 723 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBz4eiP51s37v7kf2z:-KdD4PziPUjmLYnefcCX:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L723)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

r.Context() instead of context.TODO.
</blockquote></details>

Done.

---

*[pkg/server/status.go, line 728 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBz7igszPqYN1hxygC:-KdD4QSO-y7_vLj4CET9:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L728)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

you need to return here.
</blockquote></details>

Done.

---

*[pkg/server/status.go, line 744 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBzBKvI5J4h8ttBwly:-KdD4X95vIYqdcSgGQIm:b-6e0hu3) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L744)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

why doesn't this use `ctx`?
</blockquote></details>

it should, done.

---

*[pkg/server/status.go, line 750 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBzJ2SvcI_iEI3RFdb:-KdD4lepbOykAvth-lce:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L750)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

this is a very anemic message. please consider adding some words to explain how this was determined, at least.
</blockquote></details>

Done.

---

*[pkg/server/status.go, line 815 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdBzotsVOrAjMRZRsLB:-KdD561Y64vp2Vgt9TqP:b-l3h2wc) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L815)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

why don't these use http.Error?
</blockquote></details>

They should, fixed.

---

*[pkg/server/status.go, line 818 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdC-1Fe1ZNgTpVjN8g2:-KdD57KFnLRHK418ODoR:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/status.go#L818)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

`err :=`
</blockquote></details>

Done.

---

*[pkg/server/serverpb/status.proto, line 101 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-Kd2avP5PoBhnP1bmyqk:-KdD5FsbPQpLrbrGeVBc:b-72uuv3) ([raw file](https://github.com/cockroachdb/cockroach/blob/f9744731f45b342a91b19beeb3360dfa2d8c540b/pkg/server/serverpb/status.proto#L101)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Ok, though `RangeInfo` already contains `RaftState` and other stuff, so you're probably just re-doing work. 
</blockquote></details>

My original plan was to embed a rangeInfo into rangedetail.  I'd like to add the range history to it as well.  I didn't want to add or change existing status endpoint (or their protos) without consulting with the frontend team.  But it seems clear the I should just add add the missing parts into range info.  So I'll do just that.

---

*[pkg/server/serverpb/status.proto, line 100 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdByJ6GA36gdKOiNZ1t:-KdD6H8Gt4ICEPDs8cFd:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/serverpb/status.proto#L100)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

`string error_message = 3;` - we don't use `err` in protos
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-:-KdIJ307_bFgK4rE8enl:b-9fob7f)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 10 of 10 files at r3.
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[pkg/server/status.go, line 143 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdLLah5BnMd8yCYXzVv:-KdLLah5BnMd8yCYXzVw:bajjpig) ([raw file](https://github.com/cockroachdb/cockroach/blob/6287e5b342457c7e7da6d3f8014ffe7d1dc082b3/pkg/server/status.go#L143)):*
> ```Go
> 	id, err := strconv.ParseInt(nodeIDParam, 10, 64)
> 	if err != nil {
> 		return 0, false, errors.Wrap(err, "range id could not be parsed")
> ```

node id, not range

---

*[pkg/server/status_test.go, line 559 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdLM9Faf32Cg3zSr0ur:-KdLM9Faf32Cg3zSr0us:b-wdoqzz) ([raw file](https://github.com/cockroachdb/cockroach/blob/6287e5b342457c7e7da6d3f8014ffe7d1dc082b3/pkg/server/status_test.go#L559)):*
> ```Go
> 
> 	var respError serverpb.RangeResponse
> 	if err := getStatusJSONProto(s, "range/error", &respError); err == nil {
> ```

check for the specific error.

---

*[pkg/server/status_test.go, line 572 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdLMDN2uK9Xw6yRbj0O:-KdLMDN2uK9Xw6yRbj0P:b-ebooes) ([raw file](https://github.com/cockroachdb/cockroach/blob/6287e5b342457c7e7da6d3f8014ffe7d1dc082b3/pkg/server/status_test.go#L572)):*
> ```Go
> 		t.Fatal(err)
> 	} else if !bytes.Contains(body, []byte("<TITLE>Range ID:1</TITLE>")) {
> 		t.Errorf("<title>Range Id: 1</title>, got: %s", body)
> ```

"expected"? you might want a newline before %s

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[pkg/server/status.go, line 143 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdLLah5BnMd8yCYXzVv:-KdNS6J1z2bgLzjAuoLk:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6287e5b342457c7e7da6d3f8014ffe7d1dc082b3/pkg/server/status.go#L143)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

node id, not range
</blockquote></details>

Done.

---

*[pkg/server/status_test.go, line 559 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdLM9Faf32Cg3zSr0ur:-KdNRqSTAzgNJHfpVYRY:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6287e5b342457c7e7da6d3f8014ffe7d1dc082b3/pkg/server/status_test.go#L559)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

check for the specific error.
</blockquote></details>

Done.

---

*[pkg/server/status_test.go, line 572 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdLMDN2uK9Xw6yRbj0O:-KdNRyK4bnTKSkSwUgHc:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6287e5b342457c7e7da6d3f8014ffe7d1dc082b3/pkg/server/status_test.go#L572)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

"expected"? you might want a newline before %s
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 6 of 6 files at r4.
Review status: all files reviewed at latest revision, 8 unresolved discussions, all commit checks successful.

---

*[pkg/server/status_test.go, line 561 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdN_jNdiuUkH9w_OFiK:-KdN_jNekMq2r058rpJI:bh20b1y) ([raw file](https://github.com/cockroachdb/cockroach/blob/c795d2cca6782be230d56dd0ef833b8b9ce0d8f5/pkg/server/status_test.go#L561)):*
> ```Go
> 	if err := getStatusJSONProto(s, "range/error", &respError); !testutils.IsError(err,
> 		"range id could not be parsed") {
> 		t.Errorf("expected \"range id could not be parsed\" error, got \n%s", err)
> ```

no need to change it, but why the newline here? i suggested the newline below because "body" is likely to be quite large a consist of multiple lines.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
This now uses rangeInfo instead of rangeDetail, with some additions. But it's clear that after #13587 there really isn't any need for the new status endpoint.  That seems ready to go, so I'll wait until after it's merged and then update this PR.

---

Review status: 3 of 12 files reviewed at latest revision, 8 unresolved discussions, some commit checks pending.

---

*[pkg/server/status_test.go, line 561 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdN_jNdiuUkH9w_OFiK:-KdTuHEC6dOG8yt75ciM:b3lgjcz) ([raw file](https://github.com/cockroachdb/cockroach/blob/c795d2cca6782be230d56dd0ef833b8b9ce0d8f5/pkg/server/status_test.go#L561)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

no need to change it, but why the newline here? i suggested the newline below because "body" is likely to be quite large a consist of multiple lines.
</blockquote></details>

The error in this case contains the body as well. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-:-KdTubG5ZKfOblkevx-N:b-jeefqp)*
<!-- Sent from Reviewable.io -->

petermattis commented:
Review status: 3 of 12 files reviewed at latest revision, 11 unresolved discussions, all commit checks successful.

---

*[pkg/server/server.go, line 353 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdByWPm2Xd0y9AugEz4:-KdWITqw6BXRD3vSprWk:bll0aoz) ([raw file](https://github.com/cockroachdb/cockroach/blob/a84a632833c00e7d07a3a1eeb64bb1de554faf41/pkg/server/server.go#L353)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

I've made the change.  But as I said, I'm easy going on this one, I'll change it back if there's push back on this.  My preference is to have a more stringent style guide to avoid these types of arguments.  (and the more automated it gets the better)
</blockquote></details>

Kind of silly to be flip-flopping on the style here, but this is fine for now. 

---

*[pkg/server/status.go, line 70 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdWInKrWVQ_oCTUFVtG:-KdWInKrWVQ_oCTUFVtH:b-pmq73y) ([raw file](https://github.com/cockroachdb/cockroach/blob/07b69f87428060019d1e86ed4072f012763f5129/pkg/server/status.go#L70)):*
> ```Go
> 	statusVars = statusPrefix + "vars"
> 
> 	// statusRange exposes an html page with information about a specific range.
> ```

`s/statusRange/rangeDebugEndpoint/g`

---

*[pkg/server/status.go, line 596 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdWIy6kt7XS1dflqpfB:-KdWIy6kt7XS1dflqpfC:bkw79he) ([raw file](https://github.com/cockroachdb/cockroach/blob/07b69f87428060019d1e86ed4072f012763f5129/pkg/server/status.go#L596)):*
> ```Go
> 		rep, _ := store.GetReplica(rangeID)
> 		// The only error that can be returned from getReplica is a
> 		// NewRangeNotFoundError. And we only want to return results when
> ```

`s/NewRangeNotFoundError/RangeNotFoundError/g`

---

*[pkg/server/serverpb/status.proto, line 97 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdWJTbMQ0x4BDg8Lsep:-KdWJTbMQ0x4BDg8Lseq:bhseio2) ([raw file](https://github.com/cockroachdb/cockroach/blob/07b69f87428060019d1e86ed4072f012763f5129/pkg/server/serverpb/status.proto#L97)):*
> ```protobuf
>   RaftState raft_state = 2 [(gogoproto.nullable) = false];
>   storage.storagebase.RangeInfo state = 4 [(gogoproto.nullable) = false];
>   roachpb.RangeDescriptor desc = 5 [(gogoproto.nullable) = false];
> ```

How is this different from `RangeInfo.State.Desc`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 9 of 9 files at r5.
Review status: all files reviewed at latest revision, 10 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 2 of 10 files reviewed at latest revision, 10 unresolved discussions, some commit checks pending.

---

*[pkg/server/status.go, line 70 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdWInKrWVQ_oCTUFVtG:-KdX8jeuZd8-kyvOe9wN:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/07b69f87428060019d1e86ed4072f012763f5129/pkg/server/status.go#L70)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

`s/statusRange/rangeDebugEndpoint/g`
</blockquote></details>

Done.

---

*[pkg/server/status.go, line 596 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdWIy6kt7XS1dflqpfB:-KdX8m25QoYAM9o4TRId:btm5pn) ([raw file](https://github.com/cockroachdb/cockroach/blob/07b69f87428060019d1e86ed4072f012763f5129/pkg/server/status.go#L596)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

`s/NewRangeNotFoundError/RangeNotFoundError/g`
</blockquote></details>

Remove this function.

---

*[pkg/server/serverpb/status.proto, line 97 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdWJTbMQ0x4BDg8Lsep:-KdX8oOAnMDEbKr-WDiZ:b5j3rrv) ([raw file](https://github.com/cockroachdb/cockroach/blob/07b69f87428060019d1e86ed4072f012763f5129/pkg/server/serverpb/status.proto#L97)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

How is this different from `RangeInfo.State.Desc`?
</blockquote></details>

It isn't.  Removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Ok, I think this is good to go.  PTAL

---

Review status: 2 of 10 files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-:-KdXA7nBnZAnAND18stm:b-a42h2z)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 10 of 10 files at r6.
Review status: all files reviewed at latest revision, 12 unresolved discussions, some commit checks pending.

---

*[pkg/server/status.go, line 71 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdXBakztYMS0Dgjitcf:-KdXBakztYMS0Dgjitcg:bnvcsi1) ([raw file](https://github.com/cockroachdb/cockroach/blob/6765691c2caab3937957b767d65565893f447eec/pkg/server/status.go#L71)):*
> ```Go
> 
> 	// rangeDebugEndpoint exposes an html page with information about a specific range.
> 	rangeDebugEndpoint = "/debug/range/"
> ```

seems like we don't normally include the training slash.

---

*[pkg/server/serverpb/status.proto, line 25 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdXBReGdWzfXTp9yyhy:-KdXBReGdWzfXTp9yyhz:bf2mqom) ([raw file](https://github.com/cockroachdb/cockroach/blob/6765691c2caab3937957b767d65565893f447eec/pkg/server/serverpb/status.proto#L25)):*
> ```protobuf
> import "cockroach/pkg/server/status/status.proto";
> import "cockroach/pkg/storage/engine/enginepb/mvcc.proto";
> import "cockroach/pkg/roachpb/metadata.proto";
> ```

this is unused, i think.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-:-KdXOycOnnKU_8xYtk0b:bnfp4nl)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 10 files at r3, 10 of 10 files at r6.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[pkg/server/debug.go, line 109 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdXsG9cAtPbuS8rOoNW:-KdXsG9cAtPbuS8rOoNX:b7kbnlg) ([raw file](https://github.com/cockroachdb/cockroach/blob/6765691c2caab3937957b767d65565893f447eec/pkg/server/debug.go#L109)):*
> ```Go
>       <tr>
>         <td>range status</td>
>         <td><a href="/debug/range/1">range</a></td>
> ```

If we used query arguments instead of embedding the range ID in the path, this could be a simple `<form>`. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-:-KdXwd4KvnPxL7NHEKd7:bnfp4nl)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 4 of 10 files reviewed at latest revision, 5 unresolved discussions.

---

*[pkg/server/debug.go, line 109 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdXsG9cAtPbuS8rOoNW:-KdafW3gzueazitjwx12:bw9m12g) ([raw file](https://github.com/cockroachdb/cockroach/blob/6765691c2caab3937957b767d65565893f447eec/pkg/server/debug.go#L109)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

If we used query arguments instead of embedding the range ID in the path, this could be a simple `<form>`. 
</blockquote></details>

I've changed it to take query arguments.  But haven't changed this to be a form yet.  I'll do that later.

---

*[pkg/server/status.go, line 71 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdXBakztYMS0Dgjitcf:-KdafZHEKYl3lnL6i55u:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6765691c2caab3937957b767d65565893f447eec/pkg/server/status.go#L71)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

seems like we don't normally include the training slash.
</blockquote></details>

Done.

---

*[pkg/server/serverpb/status.proto, line 25 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590#-KdXBReGdWzfXTp9yyhy:-KdaflJWVpC3MAjmoYId:braxtvx) ([raw file](https://github.com/cockroachdb/cockroach/blob/6765691c2caab3937957b767d65565893f447eec/pkg/server/serverpb/status.proto#L25)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

this is unused, i think.
</blockquote></details>

yep, removed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 6 of 6 files at r7.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13590)*
<!-- Sent from Reviewable.io -->
